src/Bounds.cpp

```diff
@@ -1185,6 +1198,19 @@ Interval bounds_of_expr_in_scope(Expr expr, const Scope<Interval> &scope, const
             << " should have been a scalar of type " << expected
             << ": " << b.interval.max << "\n";
     }
+
+    // If there is signed integer overflow, replace it with unbounded.
+    {
```

Comment:
bounds_of_expr_in_scope is used in lots of tight inner loops, and this creates a second entire recursive pass over the returned IR, so I worry about performance.

Is it not the case that signed_integer_overflow propagates upwards, and thus only the output needs to be checked to see if it equals it? Alternatively, the Bounds visitor itself could set a flag if it discovers such a call.

Reply:
> Is it not the case that signed_integer_overflow propagates upwards, and thus only the output needs to be checked to see if it equals it?

Apparently not, as the final LetStmt expression I encounter has it as just part of the expression. Will post as followup.

> Alternatively, the Bounds visitor itself could set a flag if it discovers such a call.

That sounds reasonable.

Reply:
No, I guess not. That will probably open a can of worms regarding cases where the signed integer overflow is technically unreachable, or doesn't affect the output value.

Reply:
Now that we mention it, though, it is a little curious that it isn't propagated outward here. Are there specific guidelines in the simplifier for how SIO should be propagated, or is it just ad hoc?

Reply:
It's ad-hoc. I'm really unsure as to whether it should be propagated through casts.

Reply:
Just for grins, I'm going to add that behavior to simplify for casts and see what happens.